src/main/java/org/magicdgs/readtools/utils/read/ReadWriterFactory.java

```diff
@@ -207,17 +207,20 @@ public SAMFileWriter openSAMWriter(final SAMFileHeader header, final boolean pre
             final Path output) {
         checkOutputAndCreateDirs(output);
         try {
-            return samFactory.makeWriter(header, presorted, output.toFile(), referenceFile);
+            // TODO - this will blow up if the java.nio.Path is not a file
+            // TODO - it requires an HTSJDK change not yet in the basecode (https://github.com/samtools/htsjdk/pull/1005)
+            final File referenceFile = (referencePath == null) ? null : referencePath.toFile();
```

Comment:
Suggestion for handling this: catch `UnsupportedOperationException` in the `toFile()` branch and if thrown log a warning saying that it won't be used for the output and this might fail. If the output is a CRAM file, that will fail for sure; otherwise, I believe that it is not used but better if we log a warning explaining the limitation.

Reply:
Done in another method for re-use.